### ISSUE TITLE
JSX templates are not refreshed when running eleventy --serve

### ISSUE THREAD

**Issue by dpfavand** [kaicataldo/eleventy-plugin-react#2]
Very excited about this project!

I noticed an issue when running `eleventy --serve`: the templates are never refreshed when they are changed. 

# Steps to reproduce

1. Create any page, i.e. `index.jsx` with a simple render function
```jsx
export default function Page() => { return <div> Test 1 </div> }
```
2. run `eleventy --serve`
3. View the page in the browser, note the text is "Test 1"
4. Modify `index.jsx`, change the text to "Test 2". Save, and Eleventy will indicate it rebuilt the page.
5. View the page in the browser again, note it contains the original text ("Test 1")

# Apparent Cause
In `requireFromDir`, all pages/templates are imported using `mod.require`, which is an alias for `module.require`. Once a module is required, any future attempts to require the same module will return the module as it was originally imported.

# Solution
A simple solution is to remove the module from the require cache before importing again. I've implemented a PR that does this.

I suspect my PR isn't optimal - it probably causes many more reads from disk than the original. Perhaps a list of required templates could be kept, and the cache purged only at the beginning of the build cycle.

**Comment by kaicataldo** [comment]
Hey, thanks for the PR! I hadn't tracked this bug down yet, so appreciate you digging into it. We'll definitely want to figure this out.

If we can make it performant enough, another possible solve for this is to change the mechanism currently being used to run the server-side rendered output (overloading the module system) and avoid working around the module system altogether.

**Comment by kaicataldo** [comment]
This should be fixed in v0.7.4 - confirmation that it works for you would be lovely and greatly appreciated!
